src/shared/ml-thirdparty-client/index.ts

```diff
@@ -157,4 +158,123 @@ export class Client {
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async postConsentRequests(
+    requestBody: SDKStandardComponents.PostConsentRequestsRequest,
+    destParticipantId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.postConsentRequests(requestBody)
+    }
+    await this.thirdpartyRequests.postConsentRequests(
+      requestBody,
+      destParticipantId
+    )
+  }
+
+  /**
+   * Performs a put request with authenticated consent request
+   *
+   * @param consentRequestId    unique identifier of the consent request
+   * @param requestBody         an object to authenticate consent as defined by the Mojaloop API.
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async putConsentRequests(
+    consentRequestId: string,
+    requestBody: SDKStandardComponents.PutConsentRequestsRequest,
+    destParticipantId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.putConsentRequests(consentRequestId, requestBody)
+    }
+    await this.thirdpartyRequests.putConsentRequests(
+      consentRequestId,
+      requestBody,
+      destParticipantId
+    )
+  }
+
+  /**
+   * Performs a request to generate a challenge for FIDO registration
+   *
+   * @param consentId     identifier of consent as defined by Mojaloop API.
+   */
+  public async postGenerateChallengeForConsent(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    consentId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.postGenerateChallengeForConsent(consentId)
+    }
+    // TODO: Add once implemented in sdk-standard components
+  }
+
+  /**
+   * Performs a put request with registered consent credential
+   *
+   * @param consentId     identifier of consent as defined by Mojaloop API.
+   * @param requestBody         an object to authenticate consent as defined by the Mojaloop API.
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async putConsentId(
+    consentId: string,
+    requestBody: SDKStandardComponents.PutConsentsRequest,
+    destParticipantId: string
+  ): Promise<void> {
```

Comment:
Also, why do we never want the results of these? Is there any? Do we care about those at all?

Reply:
Good point - I'm not sure actually - was just following what the rest of the repo was doing 
same with the awaiting 
so - now shifted to returning everything without await-ing and leaving it up to which ever method called this request to determine what to do with the promise
thanks!